services/venice-server/src/main/java/com/linkedin/venice/listener/ServerReadMetadataRepository.java

```diff
@@ -72,6 +77,21 @@ public MetadataResponse getMetadata(String storeName) {
                 "Fast client is not enabled for store: %s, please ensure storage node read quota is enabled for the given store",
                 storeName));
       }
+
+      if (store.isMigrating()) {
+        // only obtain store Config when store is migrating and only throw exceptions when dest cluster is ready
+        StoreConfig storeConfig = zkStoreConfigAccessor.getStoreConfig(storeName);
```

Comment:
Can we use `HelixReadOnlyStoreConfigRepository` instead of `ZkStoreConfigAccessor`? This `getStoreConfig` will generate a call to zk via `ZkBaseDataAccessor` which can be expensive when the server is handling many of these metadata requests. This approach won't generate any zk watchers but for something like StoreConfig that doesn't change frequently I think `HelixReadOnlyStoreConfigRepository` has the better trade-offs.